Refactor for clarity and control

QbikEdge opened the pull request:
- Enhanced code readability and updated documentation.
- Improved console output.
- Fixed cyclic exit: now exits after one cycle (using config.json ttl) or post-sleep.
- Introduced opt-in cyclic_config_read for periodic config reload.
- Added opt-in control for sleep_time (distinct from ttl) for dynamic adjustments.
- Minor performance optimizations.

already contains changes for [fix-wrong-ip branch](https://github.com/irfanhakim-as/cloudflare-ddns/tree/fix-wrong-ip).

QbikEdge committed:
How often does the Docker container update? #155

- added script delay to control the frequency of the script run
- changed the script so it reads the config every run. This way there is no need to restart the script after changing the config.json

QbikEdge committed:
lines correction

QbikEdge committed:
code format

QbikEdge committed:
add .vscode/ to .gitignore to exclude VS Code settings

QbikEdge committed:
Refactor for clarity and control

- Enhanced code readability and updated documentation.
- Improved console output.
- Fixed cyclic exit: now exits after one cycle (using config.json ttl) or post-sleep.
- Introduced opt-in cyclic_config_read for periodic config reload.
- Added opt-in control for sleep_time (distinct from ttl) for dynamic adjustments.
- Minor performance optimizations.

QbikEdge committed:
Merge branch 'master' into code-refactor

QbikEdge committed:
Update .gitignore

QbikEdge committed:
Update config-example.json

QbikEdge committed:
Update cloudflare-ddns.py

QbikEdge commented:
@irfanhakim-as Can you review it please

Weidav commented:
@QbikEdge would you mind publishing Docker Image based on your Fork? Merge doesn't seem to happen.

QbikEdge committed:
Enhance documentation and improve code structure

- Updated the docstring in cloudflare-ddns.py for better clarity on functionality and usage.
- Added detailed comments and structured the code for improved readability.
- Modified README.md to clarify configuration options and environment variable support.
- Removed the requests dependency from requirements.txt as it is no longer needed.
- Updated Docker build scripts to use the new image tag.
